### ISSUE TITLE
Build fails with 'Could not find AWS region, make sure it's set'

### ISSUE THREAD

**Issue by Jazzepi** [wata727/packer-post-processor-amazon-ami-management#41]
I get a `Could not find AWS region, make sure it's set.` Not sure what's going on. The account I'm using for this has full access to everything with regards to AWS. I find it odd that it prints out `    amazon-ebs (amazon-ami-management): Processing in us-east-2` which looks good to me.

Here's my packer.json
```
{
  "variables": {
    "aws_access_key": "OMITTED",
    "aws_secret_key": "OMITTED",
    "build-region": "us-east-2",
    "packer-built-this-tag": "packer-built"
  },
  "sensitive-variables": [
    "aws_secret_key"
  ],
  "builders": [
    {
      "type": "amazon-ebs",
      "access_key": "{{user `aws_access_key`}}",
      "secret_key": "{{user `aws_secret_key`}}",
      "region": "us-east-2",
      "source_ami_filter": {
        "filters": {
          "virtualization-type": "hvm",
          "name": "ubuntu/images/hvm-ssd/ubuntu-bionic-18.04-amd64-server-*",
          "root-device-type": "ebs"
        },
        "owners": [
          "099720109477"
        ],
        "most_recent": true
      },
      "subnet_filter": {
        "filters": {
          "tag:purpose": "packer-builder"
        },
        "random": true
      },
      "tags": {
        "Amazon_AMI_Management_Identifier": "{{user `packer-built-this-tag`}}"
      },
      "instance_type": "t2.micro",
      "ssh_username": "ubuntu",
      "ami_name": "ubuntu 18.04 {{timestamp}}"
    }
  ],
  "post-processors": [
    {
      "type": "amazon-ami-management",
      "regions": [
        "us-east-2"
      ],
      "identifier": "packer-built",
      "keep_releases": "3"
    }
  ]
}
```

Here's the console out
```
amazon-ebs output will be in this color.

==> amazon-ebs: Prevalidating AMI Name: ubuntu 18.04 1565815383
    amazon-ebs: Found Image ID: ami-05c1fa8df71875112
    amazon-ebs: Found Subnet ID: subnet-0b0553751a39dbc5e
==> amazon-ebs: Creating temporary keypair: packer_5d547257-0bf6-67fc-46d0-e0fa8b6aa50c
==> amazon-ebs: Creating temporary security group for this instance: packer_5d54725a-2c86-fd5a-4491-31d3f145662d
==> amazon-ebs: Authorizing access to port 22 from [0.0.0.0/0] in the temporary security groups...
==> amazon-ebs: Launching a source AWS instance...
==> amazon-ebs: Adding tags to source instance
    amazon-ebs: Adding tag: "Name": "Packer Builder"
    amazon-ebs: Instance ID: i-00861a33446c7aa96
==> amazon-ebs: Waiting for instance (i-00861a33446c7aa96) to become ready...
==> amazon-ebs: Using ssh communicator to connect: 13.58.56.59
==> amazon-ebs: Waiting for SSH to become available...
==> amazon-ebs: Connected to SSH!
==> amazon-ebs: Stopping the source instance...
    amazon-ebs: Stopping instance
==> amazon-ebs: Waiting for the instance to stop...
==> amazon-ebs: Creating AMI ubuntu 18.04 1565815383 from instance i-00861a33446c7aa96
    amazon-ebs: AMI: ami-012b028a18497998a
==> amazon-ebs: Waiting for AMI to become ready...
==> amazon-ebs: Adding tags to AMI (ami-012b028a18497998a)...
==> amazon-ebs: Tagging snapshot: snap-0ac2d1f217342e95e
==> amazon-ebs: Creating AMI tags
    amazon-ebs: Adding tag: "Amazon_AMI_Management_Identifier": "packer-built"
==> amazon-ebs: Creating snapshot tags
==> amazon-ebs: Terminating the source AWS instance...
==> amazon-ebs: Cleaning up any extra volumes...
==> amazon-ebs: No volumes to clean up, skipping
==> amazon-ebs: Deleting temporary security group...
==> amazon-ebs: Deleting temporary keypair...
==> amazon-ebs: Running post-processor: amazon-ami-management
    amazon-ebs (amazon-ami-management): Processing in us-east-2
Build 'amazon-ebs' errored: 1 error(s) occurred:

* Post-processor failed: Could not find AWS region, make sure it's set.
```

**Comment by wata727** [comment]
@Jazzepi Thank you for the report.

I confirmed the bug that occurred when configured with static credentials. I'm now working on fixing the issue.

You can avoid the bug if use environment variables or shared credentials temporarily. This is a workaround.

**Comment by wata727** [comment]
There are two problems that cause this issue.

The first one is that a region is required to start a session, which is a problem caused by old version Packer. This problem was resolved by updating the Packer dependency.

The second one is you need to pass the access key to the post-processor. Please update your configuration as follows:

```json
  "post-processors": [
    {
      "type": "amazon-ami-management",
      "access_key": "{{user `aws_access_key`}}",
      "secret_key": "{{user `aws_secret_key`}}",
      "regions": [
        "us-east-2"
      ],
      "identifier": "packer-built",
      "keep_releases": "3"
    }
  ]
```

I have confirmed that this issue is fixed in Packer 1.4.3 and the post-processor v0.7.0. Please let me know if it doesn't work. Thank you!

**Comment by Jazzepi** [comment]
Can confirm this fixed it. Thanks for the quick response!
